consensus/service/src/tx_manager.rs

```diff
@@ -110,59 +107,85 @@ pub trait UntrustedInterfaces: Clone {
     /// * `max_elements` - Maximal number of elements to output.
     ///
     /// Returns a bounded, deterministically-ordered list of transactions that are safe to append to the ledger.
-    fn combine(&self, tx_contexts: &[&WellFormedTxContext], max_elements: usize) -> Vec<TxHash>;
+    fn combine(&self, tx_contexts: &[WellFormedTxContext], max_elements: usize) -> Vec<TxHash>;
 }
 
-#[derive(Clone)]
-pub struct TxManager<
-    E: ConsensusEnclaveProxy,
-    L: Ledger,
-    UI: UntrustedInterfaces = crate::validators::DefaultTxManagerUntrustedInterfaces<L>,
-> {
-    /// Enclave.
-    enclave: E,
+#[automock]
+pub trait TxManager: Send {
```

Comment:
TxManager is now a trait, and "MockTManager" is automatically generated.

Reply:
I shortened some of the method names, and tried to make the names more closely match those of HashMap. I also replaced `missing_hashes` with `contains`, which pairs nicely with `iter().filter()` and just felt better to me.